cpp/oneapi/dal/algo/triangle_counting/backend/cpu/vertex_ranking_default_kernel_avx512.hpp

```diff
@@ -0,0 +1,942 @@
+        __m128i v_v = _mm_loadu_si128(
+            reinterpret_cast<const __m128i*>(neigh_v + i_v)); // load 8 neighbors of v
+
+        if (max_neigh_u >= max_neigh_v)
+            i_v += 4;
+        if (max_neigh_u <= max_neigh_v)
+            i_u += 4;
+
+        __mmask8 match = _mm_cmpeq_epi32_mask(v_u, v_v);
+        if (_cvtmask8_u32(match) != 0xf) { // shortcut case where all neighbors match
+            __m128i v_v1 = _mm_shuffle_epi32(v_v, _MM_SHUFFLE(0, 3, 2, 1));
+            __m128i v_v2 = _mm_shuffle_epi32(v_v, _MM_SHUFFLE(1, 0, 3, 2));
+            __m128i v_v3 = _mm_shuffle_epi32(v_v, _MM_SHUFFLE(2, 1, 0, 3));
+
+            __mmask8 tmp_match1 = _mm_cmpeq_epi32_mask(v_u, v_v1); // find matches
+            __mmask8 tmp_match2 = _mm_cmpeq_epi32_mask(v_u, v_v2);
+            __mmask8 tmp_match3 = _mm_cmpeq_epi32_mask(v_u, v_v3);
+
+            match = _kor_mask8(_kor_mask8(match, tmp_match1),
+                               _kor_mask8(tmp_match2, tmp_match3)); // combine all matches
+        }
+        total += _popcnt32_redef(_cvtmask8_u32(match)); //count number of matches
+        __m256i src = _mm256_setzero_si256();
+        __m256i one_const = _mm256_set1_epi64x(1);
+
+        __m256i gt = _mm256_mmask_i32gather_epi64(src, match, v_u, tc, 8);
+        __m256i sum = _mm256_mask_add_epi64(src, match, gt, one_const);
+        _mm256_mask_i32scatter_epi64(tc, match, v_u, sum, 8);
+    }
+    if (i_u <= (n_u - 4) && i_v < n_v) {
+        __m128i v_u = _mm_loadu_si128(reinterpret_cast<const __m128i*>(neigh_u + i_u));
+        while (neigh_v[i_v] <= neigh_u[i_u + 3] && i_v < n_v) {
+            __m128i tmp_v_v = _mm_set1_epi32(neigh_v[i_v]);
+            __mmask8 match = _mm_cmpeq_epi32_mask(v_u, tmp_v_v);
+            if (_cvtmask8_u32(match)) {
+                total++;
+                tc[neigh_v[i_v]]++;
+            }
+            i_v++;
+        }
+        i_u += 4;
+    }
+    if (i_v <= (n_v - 4) && i_u < n_u) {
+        __m128i v_v = _mm_loadu_si128(reinterpret_cast<const __m128i*>(neigh_v + i_v));
+        while (neigh_u[i_u] <= neigh_v[i_v + 3] && i_u < n_u) {
+            __m128i tmp_v_u = _mm_set1_epi32(neigh_u[i_u]);
+            __mmask8 match = _mm_cmpeq_epi32_mask(v_v, tmp_v_u);
+            if (_cvtmask8_u32(match)) {
+                total++;
+                tc[neigh_u[i_u]]++;
+            }
+            i_u++;
+        }
+        i_v += 4;
+    }
+#endif
+    while (i_u < n_u && i_v < n_v) {
+        if ((neigh_u[i_u] > neigh_v[n_v - 1]) || (neigh_v[i_v] > neigh_u[n_u - 1])) {
+            return total;
+        }
+        if (neigh_u[i_u] == neigh_v[i_v]) {
+            total++, tc[neigh_u[i_u]]++;
+            i_u++, i_v++;
+        }
+        else if (neigh_u[i_u] < neigh_v[i_v])
+            i_u++;
+        else if (neigh_u[i_u] > neigh_v[i_v])
+            i_v++;
+    }
+    return total;
+}
+
+template <typename Cpu>
+array<std::int64_t> triangle_counting_local_avx512(
```

Comment:
Why do we need `Cpu` template parameter here if even in the function name we have `avx512` suffix.
The same comment for other cpu-parametrized functions in this file.

Reply:
Fixed